server/plugin/command.go

```diff
@@ -367,9 +408,43 @@ func (p *Plugin) handleSubscribesAdd(_ *plugin.Context, args *model.CommandArgs,
 		msg += "\n\n**Warning:** You subscribed to a private repository. Anyone with access to this channel will be able to read the events getting posted here."
 	}
 
-	return msg
+	post := &model.Post{
+		ChannelId: args.ChannelId,
+		UserId:    p.BotUserID,
+		Message:   msg,
+	}
+
+	if _, appErr := p.API.CreatePost(post); appErr != nil {
+		p.API.LogWarn("error while creating post", "post", post, "error", appErr.Error())
+		return fmt.Sprintf("%s Though there was an error creating the public post: %s", msg, appErr.Error())
+	}
+
+	message := fmt.Sprintf("Successfully subscribed to [%s](%s) with events: %s.", repo, repoLink, subscribeEvents.String())
+	if previouslySubscribed {
+		message += previouslySubscribedEventMessage
+	}
+
+	return message
 }
+func (p *Plugin) fetchSubscribedEvent(args *model.CommandArgs, owner, repo string) (bool, Features, error) {
+	var previousEvents Features
+	subs, err := p.GetSubscriptionsByChannel(args.ChannelId)
+	if err != nil {
+		return false, previousEvents, err
+	}
+
+	if len(subs) == 0 {
+		return false, previousEvents, nil
+	}
 
+	for _, sub := range subs {
+		if sub.Repository == owner+"/"+repo {
+			previousEvents = Features(sub.Features)
```

Comment:
Should the  `Features` field in `Subscription` be of type `Features`?

Reply:
I'm leaving that up to you. Does the change make the code more readable?

Reply:
> @hanzei @mickmister when i subscribe to the repo then **i'm getting two bot post,** **i think we should only show 1 bot post to the user** who subscribe the repo, **we Should show the 1st bot post that visible to the all member in the channel including the member who subscribe the repo** and we should remove the "only visible to you" bot post **because why we need 2 bot post that give the same info.** pls see the screenshot ![Screenshot from 2021-12-09 01-50-06](https://user-images.githubusercontent.com/89907719/145280937-4fd1223d-2f00-498a-b8bf-c4a5c1714cd0.png)

@hanzei  @mickmister  can you let me know the view on this.

Reply:
> I'm leaving that up to you. Does the change make the code more readable? 

@hanzei can you look into the changes have made the required changes